File: frontend/src/services/feedbackService.ts
import api from './api';

export interface FeedbackItem {
  id: number;
  content: string;
  contact: string | null;
  ip: string | null;
  user_agent: string | null;
  created_at: string;
}

export interface FeedbackListResult {
  items: FeedbackItem[];
  pagination: {
    page: number;
    pageSize: number;
    total: number;
    totalPages: number;
  };
}

export const feedbackService = {
  async submit(content: string, contact?: string): Promise<void> {
    const response = await api.post('/public/feedback', {
      content,
      contact: contact || '',
    });

    if (!response.data?.success) {
      throw new Error(response.data?.error?.message || '反馈提交失败');
    }
  },

  async getAdminList(page = 1, pageSize = 20): Promise<FeedbackListResult> {
    const response = await api.get('/settings/feedback', {
      params: { page, pageSize },
    });

    if (!response.data?.success) {
      throw new Error(response.data?.error?.message || '加载反馈列表失败');
    }

    return response.data.data as FeedbackListResult;
  },
};

File: client-desktop/src/HoYoMusic.Desktop.Infrastructure/Services/FeedbackService.cs
using System.Net.Http.Json;
using HoYoMusic.Desktop.Core.Abstractions;
using HoYoMusic.Desktop.Core.Contracts;
using HoYoMusic.Desktop.Core.Models;

namespace HoYoMusic.Desktop.Infrastructure.Services;

public sealed class FeedbackService : HoYoApiClient, IFeedbackService
{
    public FeedbackService(HttpClient httpClient, ITokenStore tokenStore) : base(httpClient, tokenStore) { }

    public async Task SubmitAsync(string content, string? contact = null, CancellationToken ct = default)
    {
        using var response = await Http.PostAsJsonAsync("public/feedback", new { content, contact }, JsonOptions, ct);
        var envelope = await ReadEnvelopeAsync<object>(response, ct);
        if (!response.IsSuccessStatusCode || envelope?.Success != true)
            throw new ApiException(envelope?.Error?.Message ?? "Failed to submit feedback.", envelope?.Error?.Code);
    }

    public async Task<FeedbackListResult> GetAdminListAsync(int page = 1, int pageSize = 20, CancellationToken ct = default)
    {
        var query = new QueryStringBuilder()
            .Add("page", Math.Max(1, page))
            .Add("pageSize", Math.Clamp(pageSize, 1, 100))
            .ToString();
        return await GetAuthedAsync<FeedbackListResult>(HttpMethod.Get, $"admin/feedback{query}", "Failed to load feedback.", ct);
    }

    private async Task<TData> GetAuthedAsync<TData>(HttpMethod method, string uri, string fallbackError, CancellationToken ct) where TData : class
    {
        using var request = await CreateAuthedRequestAsync(method, uri, ct);
        using var response = await Http.SendAsync(request, ct);
        var envelope = await ReadEnvelopeAsync<TData>(response, ct);
        if (!response.IsSuccessStatusCode || envelope?.Success != true || envelope.Data is null)
            throw await CreateApiExceptionAsync(response.StatusCode, envelope?.Error?.Message ?? fallbackError, envelope?.Error?.Code, ct);
        return envelope.Data;
    }
}
